charts/flink-job/templates/podmonitor.yaml

```diff
@@ -27,4 +27,7 @@ spec:
       relabelings:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+  {{- with .Values.podMonitor.SampleLimit }}
+  SampleLimit: {{ . }}
```

Comment:
because you aren't doing anything with this, it would be a little more idiomatic to just have `SampleLimit: {{ .Values.podMonitor.sampleLimit`.

I would also put an `{{- if .Values.podMonitor.sampleLimit }} .. {{- end}}` around this, so you don't even set this to anything (meaning it would use the global default) if a value is not set in a values file.

Reply:
for my education, how is `{{- if .Values.podMonitor.sampleLimit }}` diff from `{{- with .Values.podMonitor.SampleLimit }}` and use `.` for value

Reply:
`with` defines a scope: https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with

`if` is a conditional block that does not modify scope: https://helm.sh/docs/chart_template_guide/control_structures/#ifelse